fleetshard/pkg/central/reconciler/reconciler.go

```diff
@@ -997,6 +1032,50 @@ func (r *CentralReconciler) ensurePassthroughRouteExists(ctx context.Context, re
 	return nil
 }
 
+func (r *CentralReconciler) ensureRoutesDeleted(ctx context.Context, remoteCentral private.ManagedCentral) error {
+	namespace := remoteCentral.Metadata.Namespace
+	reencryptErr := r.routeService.DeleteReencryptRoute(ctx, namespace)
+	passthroughErr := r.routeService.DeletePassthroughRoute(ctx, namespace)
+
+	if reencryptErr != nil && !apiErrors.IsNotFound(reencryptErr) { // ok if not found
+		return fmt.Errorf("deleting reencrypt route for namespace %q: %w", namespace, reencryptErr)
+	}
+	if passthroughErr != nil && !apiErrors.IsNotFound(passthroughErr) { // ok if not found
+		return fmt.Errorf("deleting passthrough route for namespace %q: %w", namespace, passthroughErr)
+	}
+	return nil
+}
+
+func (r *CentralReconciler) ensureCentralCASecretExists(ctx context.Context, centralNamespace string) (centralTLSSecretFound bool, err error) {
+	centralTLSSecretFound = true // pragma: allowlist secret
+	centralTLSSecret, err := r.getSecret(centralNamespace, k8s.CentralTLSSecretName)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			centralTLSSecretFound = false // pragma: allowlist secret
+		}
+		return centralTLSSecretFound, err
```

Comment:
Seems there is something wrong here. If the secret doesn't exist, it will never call `ensureSecretExists`. Or am I missing something?
```suggestion
		if apiErrors.IsNotFound(err) {
		  centralTLSSecretFound = false // pragma: allowlist secret
		} else {
		  return false, err
		}
```

Reply:
That's correct, but I messed up with the `if apiErrors.IsNotFound(err)` condition, it should be reversed.
I simplified this peace a little bit.
Good catch 👀 .

Reply:
The purpose of `centralTLSSecretFound` is to skip the further reconciliation instead of returning the error

Reply:
But we'll return the error in any case, so the code was correct. I simplified it a bit.